packages/utxo-global/src/ckb/index.ts

```diff
@@ -0,0 +1,142 @@
+import {
+  BytesLike,
+  KnownScript,
+  Signature,
+  SignerSignType,
+  ccc,
+} from "@ckb-ccc/core";
+import { JsonRpcTransformers } from "@ckb-ccc/core/advancedBarrel";
+import { Provider } from "../advancedBarrel";
+
+export class UtxoGlobalCKBSigner extends ccc.Signer {
+  get type(): ccc.SignerType {
+    return ccc.SignerType.CKB;
+  }
+
+  /**
+   * Gets the sign type.
+   * @returns {ccc.SignerSignType} The sign type.
+   */
+  get signType(): ccc.SignerSignType {
+    return ccc.SignerSignType.UtxoGlobalCKB;
+  }
+
+  constructor(
+    client: ccc.Client,
+    public readonly provider: Provider,
+  ) {
+    super(client);
+  }
+
+  getInternalAddress(): Promise<string> {
+    return this.getAccount();
+  }
+
+  async getIdentity(): Promise<string> {
+    return (await this.getPublicKey()).slice(2);
+  }
+
+  async getAddressObj(): Promise<ccc.Address | undefined> {
+    const address = await this.getInternalAddress();
+    return await ccc.Address.fromString(address, this.client);
+  }
+
+  async getAddressObjs(): Promise<ccc.Address[]> {
+    const address = await this.getAddressObj();
+
+    if (!!address) {
```

Comment:
if the address is empty, which means the wallet is not correctly connected, we should throw an error instead of acting like normal.

Reply:
fixed